test/whitespace/extension_types.unit

```diff
@@ -0,0 +1,175 @@
+40 columns                              |
+>>> indentation
+extension type const A(int a) {
+inc(int x) => ++x;
+foo(int x) {
+if (x == 0) {
+return true;
+}}}
+<<<
+extension type const A(int a) {
+  inc(int x) => ++x;
+  foo(int x) {
+    if (x == 0) {
+      return true;
+    }
+  }
+}
+>>> all tokens
+@Anno<int, int>() extension type const A<S, T>.name(@required Map<int, int> a) implements I<S>, J<T> {}
+<<<
+@Anno<int, int>()
+extension type const A<S, T>.name(
+    @required Map<int, int> a)
+    implements I<S>, J<T> {}
```

Comment:
This doesn't look great to me. There's an implicit principle in the formatter that adjacent lines shouldn't have the same indentation but for different reasons. In this case, line three is implemented +4 because of the representation type's parameter list and the next line is indented +4 because it's a split `implements` clause.

If the representation type splits, I would indent it past the `implements` clause, like:

```dart
@Anno<int, int>()
extension type const A<S, T>.name(
        @required Map<int, int> a)
    implements I<S>, J<T> {}
```

(Eventually, with the new style, it will look more like, maybe:

```dart
@Anno<int, int>()
extension type const A<S, T>.name(
  @required Map<int, int> a,
) implements I<S>, J<T> {}
```

)

To implement this, try moving:

```dart
builder.startRule(CombinatorRule());
```

Above:

```dart
    visit(node.typeParameters);
    visit(node.representation);
```

And then you might need to wrap another `nestExpression()` `unnest()` pair around:

```dart
    builder.startRule(CombinatorRule());
    visit(node.implementsClause);
    builder.endRule();
```

(I could be wrong about this. There tends to be some trial and error with hacking on the formatter. That's one of the reasons I want to move to a better internal representation.)

Reply:
We want the "primary construtor parameter" to be indented more *iff* there is a following `implements` clause, and they are all on separate lines.

I seem to have succeeded in that using:
```dart
    builder.nestExpression();
    visit(node.typeParameters);
    visit(node.representation);
    builder.unnest();

    builder.startRule(CombinatorRule());
    visit(node.implementsClause);
    builder.endRule();
```